src/openvic-simulation/map/ProvinceInstance.cpp

```diff
@@ -35,6 +38,18 @@ ProvinceInstance::ProvinceInstance(
 	religion_distribution {},
 	max_supported_regiments { 0 } {}
 
+GoodDefinition const* ProvinceInstance::get_rgo_good() const {
+	if(!rgo.is_valid()) { return nullptr; }
+	return &(rgo.get_production_type()->get_output_good());
+}
+void ProvinceInstance::set_rgo_production_type(ProductionType const& rgo_production_type) {
+	if(rgo_production_type.get_template_type() != ProductionType::template_type_t::RGO) {
+		//error
```

Comment:
Sure, it's a scenario that should never happen though.
When changing rgo via event, the event parser should catch this error instead of the rgo.
Same thing with start dates.
I'll make it return a bool and log error anyway (this code was before I realised we have a global logger).